<center><h1>Técnicas de Inteligencia Artificial</header1></center>

Presentado por: Nombres Apellidos  <br>
Fecha: DD/MM/2022

# Importación de librerias necesarias

In [1]:
#Para esta actividad se importarán las siguientes librerías:
import pandas as pd

## Cargar el Dataset

In [2]:
#Código para cargar el Dataset
df = pd.read_csv('Realestate.csv')
df

No  X1 transaction date  X2 house age  \
0      1             2012.917          32.0   
1      2             2012.917          19.5   
2      3             2013.583          13.3   
3      4             2013.500          13.3   
4      5             2012.833           5.0   
..   ...                  ...           ...   
409  410             2013.000          13.7   
410  411             2012.667           5.6   
411  412             2013.250          18.8   
412  413             2013.000           8.1   
413  414             2013.500           6.5   

     X3 distance to the nearest MRT station  X4 number of convenience stores  \
0                                  84.87882                               10   
1                                 306.59470                                9   
2                                 561.98450                                5   
3                                 561.98450                                5   
4                                 390.56840                                5   
..                                      ...                              ...   
409                              4082.01500                                0   
410                                90.45606                                9   
411                               390.96960                                7   
412                               104.81010                                5   
413                                90.45606                                9   

     X5 latitude  X6 longitude  Y house price of unit area  
0       24.98298     121.54024                        37.9  
1       24.98034     121.53951                        42.2  
2       24.98746     121.54391                        47.3  
3       24.98746     121.54391                        54.8  
4       24.97937     121.54245                        43.1  
..           ...           ...                         ...  
409     24.94155     121.50381                        15.4  
410     24.97433     121.54310                        50.0  
411     24.97923     121.53986                        40.6  
412     24.96674     121.54067                        52.5  
413     24.97433     121.54310                        63.9  

[414 rows x 8 columns]

## Descripción de la fuente del Dataset

El dataset contiene información de 414 viviendas, dicho dataset contiene 8 atributos, incluida una característica de clase objetivo que indica el precio de la vivienda. El significado de cada columna en los datos es el siguiente:<br><br>
**X1 transaction date**: Continuo. Fecha en la que se negocia.<br>
**X2 house age**: Discreta. Edad de la casa.<br>
**X3 distance to the nearest MRT station**: Continuo. La estación de MRT más cercana.<br>
**X4 number of convenience stores**: Discreta. Número de tiendas de conveniencia.<br>
**X5 latitude**: Continuo. Latitud.<br>
**X6 longitude**: Continuo. Longitud.<br>
**Y house price of unit area**: Continuo. Precio de la vivienda por unidad de superficie.<br>

## Explique el problema a resolver. 
Se requiere crear un modelo de aprendizaje automático que permita predecir el precio adecuado de inmobiliarios en función de su ubicación. Para lo cual se utilizara el dataset (Real estate), el cual contiene 414 viviendas. Donde, la variable objetiva se denomina **”Y house price of unit area”** . Existen 7 variables de entrada llamadas: **No, X1 transaction date, X3 distance to the nearest MRT station,X4 number of convenience stores, X5 latitude, X6 longitude.**

## Caracterización del Dataset

Realice una descripción de los datos con:

>- Número de instancias en total.
>- Número de atributos de entrada, su significado y tipo.
>- Estadísticas de la variable objetivo.
>- Estadísticas los atributos en relación con la variable objetivo.


## Número de instancias en total.

In [3]:
#Número de instancias en total
df.count()

No                                        414
X1 transaction date                       414
X2 house age                              414
X3 distance to the nearest MRT station    414
X4 number of convenience stores           414
X5 latitude                               414
X6 longitude                              414
Y house price of unit area                414
dtype: int64

### Número de atributos de entrada, su significado y tipo

**X1 transaction date**: Continuo. Fecha en la que se negocia.<br>
**X2 house age**: Discreta. Edad de la casa.<br>
**X3 distance to the nearest MRT station**: Continuo. La estación de MRT más cercana.<br>
**X4 number of convenience stores**: Discreta. Número de tiendas de conveniencia.<br>
**X5 latitude**: Continuo. Latitud.<br>
**X6 longitude**: Continuo. Longitud.<br>

In [6]:
df.drop('Y house price of unit area', axis=1).info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 414 entries, 0 to 413
Data columns (total 7 columns):
 #   Column                                  Non-Null Count  Dtype  
---  ------                                  --------------  -----  
 0   No                                      414 non-null    int64  
 1   X1 transaction date                     414 non-null    float64
 2   X2 house age                            414 non-null    float64
 3   X3 distance to the nearest MRT station  414 non-null    float64
 4   X4 number of convenience stores         414 non-null    int64  
 5   X5 latitude                             414 non-null    float64
 6   X6 longitude                            414 non-null    float64
dtypes: float64(5), int64(2)
memory usage: 22.8 KB


## Estadísticas de la variable objetivo.

In [5]:
df[['Y house price of unit area']].describe()

Y house price of unit area
count                  414.000000
mean                    37.980193
std                     13.606488
min                      7.600000
25%                     27.700000
50%                     38.450000
75%                     46.600000
max                    117.500000

## Estadísticas los atributos en relación con la variable objetivo.

In [7]:
df[[	'X1 transaction date',	'X2 house age',	'X3 distance to the nearest MRT station',	'X4 number of convenience stores',	'X5 latitude',	'X6 longitude']].describe()

X1 transaction date  X2 house age  \
count           414.000000    414.000000   
mean           2013.148971     17.712560   
std               0.281967     11.392485   
min            2012.667000      0.000000   
25%            2012.917000      9.025000   
50%            2013.167000     16.100000   
75%            2013.417000     28.150000   
max            2013.583000     43.800000   

       X3 distance to the nearest MRT station  \
count                              414.000000   
mean                              1083.885689   
std                               1262.109595   
min                                 23.382840   
25%                                289.324800   
50%                                492.231300   
75%                               1454.279000   
max                               6488.021000   

       X4 number of convenience stores  X5 latitude  X6 longitude  
count                       414.000000   414.000000    414.000000  
mean                          4.094203    24.969030    121.533361  
std                           2.945562     0.012410      0.015347  
min                           0.000000    24.932070    121.473530  
25%                           1.000000    24.963000    121.528085  
50%                           4.000000    24.971100    121.538630  
75%                           6.000000    24.977455    121.543305  
max                          10.000000    25.014590    121.566270

## Preprocesamiento del dataset. Transformaciones previas necesarias para la modelación

In [8]:
#Transformación de las caracteristicas (mire el apartado Feature engineering del aula virtual)
# Variable objetivo y variables de entrada
VariableObjetivo=['Y house price of unit area']
VariablesEntrada=[	'X1 transaction date',	'X2 house age',	'X3 distance to the nearest MRT station',	'X4 number of convenience stores',	'X5 latitude',	'X6 longitude']

In [9]:
X=df[VariablesEntrada].values
y=df[VariableObjetivo].values

In [10]:
### Estandarización de la data ###
from sklearn.preprocessing import StandardScaler
VariablesEntradaScaler=StandardScaler()
VariableObjetivoScaler=StandardScaler()

In [11]:
#Almacenamiento del objeto de ajuste para referencia posterior
VariablesEntradaFit=VariablesEntradaScaler.fit(X)
VariableObjetivoFit=VariableObjetivoScaler.fit(y)

In [12]:
#Generación de los valores estandarizados de X e y
X=VariablesEntradaFit.transform(X)
y=VariableObjetivoFit.transform(y)

## División del dataset en datos de entrenamiento y datos de test 

In [5]:
#Código que realice la división en entrenamiento y test, de acuerdo con la estretgia de evaluación planeada. Describa cuál es.


## Regresión Lineal

In [6]:
#Códo del modelo

In [7]:
#Entrenamiento

In [8]:
#Test

## Evaluación del Modelo

Construya un o dos párrafos con los principales hallazgos. Cómo está funcionando el modelo? Calidad en los resultados de predicción. 